File: src/main/java/org/nia/logic/quests/IQuestEvent.java
package org.nia.logic.quests;

import org.nia.model.Quest;

/**
 * @author IANazarov
 */
public interface IQuestEvent {

    String getName();

    default void init(Quest quest) { }

    IQuestStep getInit();
    default boolean canBeRandomed() {
        return true;
    }

    IQuestStep getQuestStep(String questStep);

    default int getReward() {
        return 7;
    }
    QuestsEnum getQuestsEnum();
}

File: src/main/java/ru/nia/tavern/quests/IQuestEvent.kt
package ru.nia.tavern.quests

import ru.nia.tavern.model.Quest


/**
 * @author IANazarov
 */
interface IQuestEvent {

    val name: String

    val init: IQuestStep

    val reward: Int
        get() = 6
    val questsEnum: QuestsEnum

    open fun init(quest: Quest) {}
    open fun canBeRandomed(): Boolean {
        return true
    }

    fun getQuestStep(questStep: String): IQuestStep
}
